### PR TITLE
Stackrox: RS-532 ci job template simplify

### PR DESCRIPTION
**Pull request by davdhacs**
open the gate: gke-qa-e2e-tests
open the gate: gke-upgrade-tests
/hold
/cc house-d

### CODE REVIEW

**Comment by gavin-stackrox** [comment]
It is a pity that YAML anchors/aliases are not supported in ci-operator/config files.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ postsubmits:
       timeout: 3h0m0s
     labels:
       ci.openshift.io/generator: prowgen
-    max_concurrency: 5
+    max_concurrency: 1
```
[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml:14]: FWIW `max_concurrency: 5` was something I added earlier today (5/24).

[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml]: > What do you think of this pattern?

I think its fantastic!. So much better. And the JOB_NAME convention seems fine to me. Maybe `${TEST_SUITE:-${JOB_NAME...` at some future point if needed. I'll do a proper review in the morning. Cheers!

**Comment by gavin-stackrox** [comment]
Yes, I think `<org>-<flavor>` will be useful. And seems the easiest to manage for the size of our org. In CIs current form there are a lot of shared credentials used by scanner/collector/stackrox CI jobs. So I think for the purposes of this PR 3 org level flavors will work: initial, e2e & push.

Regarding the env to use to get the 'job' I think `JOB_NAME_SAFE` is good to use. While it is not [documented](https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables) as a supported var that I could find it is used throughout other openshift/release configs, so its er safe and seems to be correct for pj-rehearse jobs also.

It would be good to test this with a config for a branch in stackrox/stackrox i.e. a branch there called `house-RS-532-ci-job-template-simplify` so that we can see it run the e2e and style jobs.

**Comment by gavin-stackrox** [comment]
actually... the branch in stackrox/stackrox was probably not necessary because there will probably not be any changes needed to be made, so testing against master is fine. Sorry :(

**Review by gavin-stackrox — APPROVED** [review]
LGTM. Even though the upgrade test failed, it still shows that the creds were passed correctly, so I think this is working as desired.

**Comment by vjwilson** [comment]
Test comment

Does this link work?

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/28832/rehearse-28832-pull-ci-stackrox-stackrox-house-RS-532-ci-job-template-simplify-gke-qa-e2e-tests/1529563639739060224

**Review by gavin-stackrox — APPROVED** [review]
/lgtm

**Pull request merged**